docker/docker-entrypoint.sh

```diff
@@ -52,40 +52,55 @@ typeset -a TUNNEL_OPTS
-        if [[ -n "${JWT_FILE:-}" ]]; then
-            echo "INFO: enrolling ${JWT_FILE}"
-            ziti-edge-tunnel enroll --jwt "${JWT_FILE}" --identity "${IDENTITY_FILE}" || {
-                echo "ERROR: failed to enroll with token from ${JWT_FILE} ($(wc -c < "${JWT_FILE}")B)" >&2
-                exit 1
-            }
-        elif [[ -n "${NF_REG_TOKEN:-}" ]]; then
-            echo "INFO: attempting enrollment with NF_REG_TOKEN"
-            ziti-edge-tunnel enroll --jwt - --identity "${IDENTITY_FILE}" <<< "${NF_REG_TOKEN}" || {
-                echo "ERROR: failed to enroll with token from NF_REG_TOKEN ($(wc -c <<<"${NF_REG_TOKEN}")B)" >&2
-                exit 1
-            }
+        echo "ERROR: need integer for NF_REG_WAIT" >&2
+        exit 1
+    fi
+    while (( $NF_REG_WAIT > 0 || $NF_REG_WAIT < 0)); do
+        # if non-empty identity file
+        if [[ -s "${IDENTITY_FILE}" ]]; then
+            echo "INFO: found identity file ${IDENTITY_FILE}"
+            break 1
+        # look for enrollment token
         else
-            echo "INFO: ${NF_REG_NAME}.jwt was not found, trying stdin" >&2
-            ziti-edge-tunnel enroll --jwt - --identity "${IDENTITY_FILE}" || {
-                echo "ERROR: failed to enroll with token from stdin" >&2
-                exit 1
-            }
+            echo "INFO: identity file ${IDENTITY_FILE} does not exist"
+            for dir in "/var/run/secrets/netfoundry.io/enrollment-token" "${IDENTITIES_DIR}"; do
+                JWT_CANDIDATE="${dir}/${NF_REG_NAME}.jwt"
+                echo "INFO: looking for ${JWT_CANDIDATE}"
+                if [[ -s "${JWT_CANDIDATE}" ]]; then
+                    JWT_FILE="${JWT_CANDIDATE}"
+                    break 1
+                fi
+            done
+            if [[ -n "${JWT_FILE:-}" ]]; then
+                echo "INFO: enrolling ${JWT_FILE}"
+                ziti-edge-tunnel enroll --jwt "${JWT_FILE}" --identity "${IDENTITY_FILE}" || {
+                    echo "ERROR: failed to enroll with token from ${JWT_FILE} ($(wc -c < "${JWT_FILE}")B)" >&2
+                    exit 1
+                }
+            elif [[ -n "${NF_REG_TOKEN:-}" ]]; then
+                echo "INFO: attempting enrollment with NF_REG_TOKEN"
+                ziti-edge-tunnel enroll --jwt - --identity "${IDENTITY_FILE}" <<< "${NF_REG_TOKEN}" || {
+                    echo "ERROR: failed to enroll with token from NF_REG_TOKEN ($(wc -c <<<"${NF_REG_TOKEN}")B)" >&2
+                    exit 1
+                }
+            elif [[ -n "${NF_REG_STDIN:-}" ]]; then
+                echo "INFO: trying to get token from stdin" >&2
+                ziti-edge-tunnel enroll --jwt - --identity "${IDENTITY_FILE}" || {
+                    echo "ERROR: failed to enroll with token from stdin" >&2
+                    exit 1
+                }
+            fi
         fi
-    fi
+        # decrement the wait seconds until zero or forever if negative
+        let NF_REG_WAIT--
```

Comment:
so technically this will be "kinda sorta" the number of seconds - which is fine for me. you could use the actual clock time instead to be more accurate if you really care... i doubt we really care though?

Reply:
You're correct. More accurately, it's "at least N seconds", and the realtime will always be a tiny fraction of 1s more than the max N seconds specified.